Binary files bfgminer-3.10.10.orig/bfgminer and bfgminer-3.10.10/bfgminer differ
Binary files bfgminer-3.10.10.orig/bfgminer-boinc_wrapper.o and bfgminer-3.10.10/bfgminer-boinc_wrapper.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-deviceapi.o and bfgminer-3.10.10/bfgminer-deviceapi.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-driver-cpu.o and bfgminer-3.10.10/bfgminer-driver-cpu.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-driver-opencl.o and bfgminer-3.10.10/bfgminer-driver-opencl.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-miner.o and bfgminer-3.10.10/bfgminer-miner.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-scrypt.o and bfgminer-3.10.10/bfgminer-scrypt.o differ
Binary files bfgminer-3.10.10.orig/bfgminer-util.o and bfgminer-3.10.10/bfgminer-util.o differ
Only in bfgminer-3.10.10: boinc_lockfile
diff -ur -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/boinc_wrapper.cpp bfgminer-3.10.10/boinc_wrapper.cpp
--- bfgminer-3.10.10.orig/boinc_wrapper.cpp	2016-01-13 16:06:41.750916787 -0800
+++ bfgminer-3.10.10/boinc_wrapper.cpp	2016-01-13 18:10:43.972862703 -0800
@@ -2,16 +2,24 @@
 #include "boinc_wrapper.h"
 
 #include "boinc_api.h"
+#include "diagnostics.h"
+//#include "graphics2.h"
 #include <filesys.h>
 #include <string>
 
 using namespace std;
 
 BOINC_OPTIONS options;
+typedef struct _BW_SHMEM {
+	int nothing;
+} BW_SHMEM;
+
+APP_INIT_DATA app_init_data;
+BW_SHMEM *bw_shmem;
 
 void bw_boinc_begin_critical_section()
 {
-	bw_boinc_begin_critical_section();
+	boinc_begin_critical_section();
 }
 
 void bw_boinc_end_critical_section()
@@ -21,7 +29,7 @@
 
 void bw_boinc_fraction_done(double x)
 {
-	bw_boinc_fraction_done(x);
+	boinc_fraction_done(x);
 }
 
 int bw_boinc_get_status_suspended()
@@ -31,26 +39,40 @@
 	return bs.suspended;
 }
 
-void bw_boinc_init()
-{
-	//boinc_init(); //init simple program
-}
-
 void bw_boinc_finish(int x)
 {
 	boinc_finish(x);
 }
 
-void bw_boinc_init_options()
+void bw_boinc_init()
 {
+	int retval = 0;
+	unsigned long dwDiagnosticsFlags = 0;
+	retval = boinc_init_diagnostics(dwDiagnosticsFlags);
+	boinc_parse_init_data_file();
+	boinc_get_init_data(app_init_data);
+	if (!strlen(app_init_data.project_dir)) {
+#ifdef _WIN32
+	    strcpy(app_init_data.project_dir, ".\\");
+#else
+	    strcpy(app_init_data.project_dir, "./");
+#endif
+	}
+
+	//bw_shmem = (BW_SHMEM*)boinc_graphics_make_shmem("bfgminer", sizeof(BW_SHMEM));
+	//if (!bw_shmem) {
+        //	bw_shmem=(BW_SHMEM*)boinc_graphics_get_shmem("bfgminer");
+	//}
+
 	//more complex init
 	boinc_options_defaults(options);
         options.normal_thread_priority = true;
 	options.check_heartbeat = false;
 	options.handle_process_control = true; //program handles suspend,quit,resume
-	options.direct_process_action = false; //quit and heartbeat exit, suspend=suspend, resume=resume
+	options.direct_process_action = false; //true means quit and heartbeat exit, suspend=suspend, resume=resume; false means set the flag in boinc_status
 	options.send_status_msgs = true; //report cpu time and fraction done progress
-	boinc_init_options(&options);
+	boinc_init_options(&options); //for complicated apps
+	//boinc_init_options(); //for simple apps
 }
 
 void bw_boinc_show_message(int status, const char *msg, int is_notice)
diff -ur -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/boinc_wrapper.h bfgminer-3.10.10/boinc_wrapper.h
--- bfgminer-3.10.10.orig/boinc_wrapper.h	2016-01-13 16:06:41.747916672 -0800
+++ bfgminer-3.10.10/boinc_wrapper.h	2016-01-13 18:09:10.758259286 -0800
@@ -15,7 +15,6 @@
 int bw_boinc_get_status_suspended();
 void bw_boinc_init();
 void bw_boinc_finish(int x);
-void bw_boinc_init_options();
 void bw_boinc_show_message(int status, const char *msg, int is_notice);
 FILE* bw_boinc_fopen(char *path, char *mode);
 void bw_boinc_fclose(FILE *file);
Binary files bfgminer-3.10.10.orig/ccan/libccan.a and bfgminer-3.10.10/ccan/libccan.a differ
Binary files bfgminer-3.10.10.orig/lib/libgnu.a and bfgminer-3.10.10/lib/libgnu.a differ
diff -ur -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/miner.c bfgminer-3.10.10/miner.c
--- bfgminer-3.10.10.orig/miner.c	2016-01-13 16:06:41.750916787 -0800
+++ bfgminer-3.10.10/miner.c	2016-01-13 18:08:44.553246200 -0800
@@ -9991,8 +9991,6 @@
 	memset(&zero_tv, 0, sizeof(struct timeval));
 	cgtime(&rotate_tv);
 
-
-
 	while (1) {
 		int i;
 		struct timeval now;
@@ -10025,7 +10023,6 @@
 
 
 #ifdef USE_BOINC
-
 		if (!sched_paused && (suspended_curr != suspended_prev) && suspended_curr) {
 			sched_paused = true;
 
@@ -11366,8 +11363,6 @@
 
 #ifdef USE_BOINC
 	bw_boinc_init();
-
-	bw_boinc_init_options();
 #endif
 
 #ifdef WIN32
diff -ur -x configure -x config.status -x config.log -x aclocal.m4 -x Makefile -x Makefile.in bfgminer-3.10.10.orig/miner.h bfgminer-3.10.10/miner.h
--- bfgminer-3.10.10.orig/miner.h	2016-01-13 16:06:41.743916518 -0800
+++ bfgminer-3.10.10/miner.h	2016-01-13 17:36:41.654203710 -0800
@@ -755,11 +755,6 @@
 
 #define flip32(dest_p, src_p) swap32yes(dest_p, src_p, 32 / 4)
 
-#ifdef USE_BOINC
-void boinc_quit_status(int status, const char *format, ...);
-void boinc_work_success();
-#endif
-
 #define WATCHDOG_INTERVAL  2
 extern void bfg_watchdog(struct cgpu_info *, struct timeval *tvp_now);
 
@@ -1565,5 +1560,11 @@
 extern struct api_data *api_add_diff(struct api_data *root, char *name, double *data, bool copy_data);
 extern struct api_data *api_add_json(struct api_data *root, char *name, json_t *data, bool copy_data);
 
+#ifdef USE_BOINC
+void boinc_quit_status(int status, const char *format, ...);
+void boinc_work_success();
+#endif
+
+
 
 #endif /* __MINER_H__ */
Only in bfgminer-3.10.10: phatk121016Juniperv2w128l8pOpenCL2_0AMDAPP1800_11.bin
Only in bfgminer-3.10.10: result.txt
Binary files bfgminer-3.10.10.orig/x86_64/libx8664.a and bfgminer-3.10.10/x86_64/libx8664.a differ
